crates/broker/src/market_monitor.rs

```diff
@@ -282,14 +317,24 @@ where
             .into_stream()
             .for_each(|log_res| async {
                 match log_res {
-                    Ok((event, _)) => {
+                    Ok((event, log)) => {
                         tracing::info!("Detected request fulfilled {:x}", event.requestId);
-                        if let Err(e) = db.set_order_complete(U256::from(event.requestId)).await {
-                            tracing::error!("Failed to update order status to Done: {e:?}");
+                        if let Err(e) = db
+                            .set_request_fulfilled(
```

Comment:
Do we need to distinguish here on whether or not the prover was paid? Two cases where this distinction may matter if when someone fulfills an order we already locked (i.e. they don't get paid, and we are still eligible for payment), and if the order is not locked and someone else fills it but doesn't get paid due to client not having enough funds (https://github.com/boundless-xyz/boundless/issues/545). The latter of the two edge cases may be niche enough to ignore.

Reply:
That would be useful to store

I'm inclined to leave this to a follow up as 1/ Currently the last point at which we consider aborting a fulfillment is right before we lock it, so handling these cases would be new code. and 2/ Its actually quite tricky to get this info at the moment since the payment failed event doesn't include the request id